src/utils/utils.py

```diff
@@ -0,0 +1,235 @@
+            return True
+    return False
+
+def get_med2(row):
+    temp = t_med2[t_med2["HADM_ID"]==row["HADM_ID"]] 
+    return temp[temp["ITEMID"]==row["ITEMID"]].iloc[0]
+from sklearn import datasets, linear_model, metrics
+
+def get_normalized_trend(data):
+    selected = data[['VALUENUM', 'hours_from_med']]
+    if selected.shape[0]<2:
+        return float("NaN")
+    reg = linear_model.LinearRegression()
+    reg.fit(np.array(data['hours_from_med']).reshape(-1,1), np.array(data['VALUENUM']).reshape(-1,1))
+    return reg.coef_[0][0]
+
+def get_normalized_trend_np(data):
+    selected = data[['VALUENUM', 'hours_in']]
+    print(selected)
+    if selected.shape[0]<2:
+        return float("NaN")
+    print(np.array(data['hours_from_med']), np.array(data['VALUENUM']))
+    t = np.polyfit(np.array(data['hours_from_med']), np.array(data['VALUENUM']), 1,full=True)
+    coefficients, residuals, _, _, _ = t
+    print(t)
+    mse = residuals[0]/(len(selected.index))
+    nrmse = np.sqrt(mse)/(selected.max() - selected.min())
+    return 1
+
+# Utils class
+class AnalysisUtils:
+
+    def __init__(self, data, res, gender="MF", age_b=0, age_a=100, ethnicity="WHITE", lab_mapping=None, load=False):
+        '''
+        Params
+        data : path to dataset
+        res : path to result output files
+        gender : stratification param for gender
+        age_b : stratification param for start of age group
+        age_a : stratification param for end of age group
+        ethnicity : stratification param for ethnicity
+        lab_mapping : lab test mapping from mimic extract. Loaded externally and used in the class 
+        '''
+        self.data = data 
+        self.res = res
+        self.gender = gender
+        self.age_b = age_b
+        self.age_a = age_a
+        self.ethnicity = ethnicity
+        self.stratify_prefix = f"{age_b}-{age_a}_{gender}_{ethnicity}"
+
+        self.res_dict_mapping_med = None
+        self.d_m_l_doc = None
+        if load:
+            self.load_mappings()
+        self.lab_mapping = LAB_MAPPING
+
+    def load_mappings(self):
+        """
+        Load Medication and Lab test name mappings from MIMIC Extract and Clinically Validated sources
+        """
+        self.d_m_l_doc = pd.read_csv(os.path.join(self.data, "mimiciii", "1.4","preprocessed", "mapping_med_itemid_doc.csv")).drop(columns=["Unnamed: 0"])
+        dict_d_m_l = self.d_m_l_doc.to_dict("records")
+        self.res_dict_mapping_med = {
+            v:k["Medication"] for k in dict_d_m_l for v in [int(id) for id in k["ITEMID_with_manual"][1:-1].split(",") if id != '']
+        }
+
+    def generate_med_lab_pairs(self):
+        """
+        Generate medication and lab test pair names.
+        """
+        
+        d_lab_map = {k:list(v.keys()) for k, v in self.lab_mapping.items()}
+        indexes = list(self.d_m_l_doc.groupby(["Medication", "lab result"]).count().index)
```

Comment:
That's not working, `d_m_l_doc` has no cols with the names `["Medication", "lab result"]`

Reply:
Removed `d_m_l_doc` dependency.